Lib/test/test_typing.py

```diff
@@ -894,6 +906,93 @@ class C(Generic[Unpack[Ts]]): pass
         self.assertNotEqual(C[Unpack[Ts1]], C[Unpack[Ts2]])
 
 
+class TypeVarTuplePicklingTests(BaseTestCase):
+    # These are slightly awkward tests to run, because TypeVarTuples (and things
+    # made generic using TypeVarTuples) are only picklable if defined in the
+    # global scope. We therefore need to push various things defined in these
+    # tests into the global scope with `global` statements at the start of each
+    # test.
+
+    @all_pickle_protocols
+    def test_pickling_then_unpickling_results_in_equality(self, proto):
+        global Ts1  # See explanation at start of class.
+        Ts1 = TypeVarTuple('Ts1')
+        Ts2 = pickle.loads(pickle.dumps(Ts1, proto))
+        self.assertEqual(Ts1, Ts2)
+
+    @all_pickle_protocols
+    def test_pickling_then_unpickling_unpacked_results_in_equality(self, proto):
+        global Ts  # See explanation at start of class.
+        Ts = TypeVarTuple('Ts')
+        unpacked1 = Unpack[Ts]
+        unpacked2 = pickle.loads(pickle.dumps(unpacked1, proto))
+        self.assertEqual(unpacked1, unpacked2)
+
+    @all_pickle_protocols
+    def test_pickling_then_unpickling_variadic_class_results_in_equality(
+            self, proto
+    ):
+        global T, Ts, A1, B1  # See explanation at start of class.
```

Comment:
This misses the point a bit. Classes are always pickled by name, so it doesn't really matter what Ts is doing here. I think we can simply remove this test.

Reply:
Done.